src/app/forms/services/dynamic-form.service.ts

```diff
@@ -62,13 +62,13 @@ export class DynamicFormService {
 
   createControls(child: FormNode, d: any) {
     const controls: any[] = [];
-    if (d.length && d.length > 0) {
+    if (Array.isArray(d)) {
```

Comment:
This is a clever check because if is `d` is an `Array`, it won't matter if it's empty or not. Nice 👌